project-management/src/main/java/life/qbic/projectmanagement/application/ProjectCreationService.java

```diff
@@ -97,10 +97,15 @@ private Project createProject(String code, String title, String objective,
 
     var registrationResult = projectDomainService.registerProject(intent, projectCode, projectManager, principalInvestigator, responsiblePerson, funding);
 
-    if (registrationResult.isError() && registrationResult.getError().equals(ResponseCode.PROJECT_REGISTRATION_FAILED)) {
-      throw new ApplicationException("Project registration failed.", ErrorCode.GENERAL, ErrorParameters.of(code));
-    }
-
+    registrationResult.onError(responseCode -> {
+      if (responseCode.equals(ResponseCode.PROJECT_ALREADY_EXISTS)) {
+        throw new ApplicationException("Project code is already in use.",
```

Comment:
```suggestion
        throw new ApplicationException("Project code " + projectCode +" is already in use.",
```
you can add debug information here. This message is logged.